index.js

```diff
@@ -393,4 +533,64 @@ function isaReplyTo (a, b) {
 
 function isaMentionTo (a, b) {
   return mlib.relationsTo(a, b).indexOf('mentions') >= 0
+}
+
+function isaRevisionTo (a, b) {
+  var rels = mlib.relationsTo(a, b)
+  return rels.indexOf('revision') >= 0
+}
+
+function removeThreadDuplicates(threadArr) {
+  // remove duplicates by converting keys into a set
+  const uniqKeys = new Set(threadArr.map(function(t) { return t.key}))
+  var uniqFlatThread = []
+  
+  for (var item in uniqKeys) {
+    uniqFlatThread.push(threadArr.find(function(t) { return t.key === uniqKeys[item]}))
+  }
+  return uniqFlatThread
+}
+
+function pluckRecursively( a, prop, mem ){
+  mem = mem || []
+  if( a[prop] ){
+    mem.push(a[prop])
+    pluckRecursively( a[prop], prop, mem )
+  }
+  return mem
+}
+
+function isNewerRevision(msg, rev) {
+  // more explicit check of revision relationship than above
+  var root = mlib.link(rev.root)
+  const relMsg       = (rev.value.content)
+  const isEdit       = (relMsg.type === 'post-edit')
+  const editsThis    = (relMsg.revision === msg.key)
+  const sameAuthor   = (rev.value.author === msg.value.author)
+  const isWiki       = (rev.value.author === 'wiki')
```

Comment:
Yeah, I appreciate the idea but let's remove it. There's a lot of complexity around multiple-user editing in EC systems, which we cant handle yet, and wouldnt handle this way.